src/thunderbots/backend_output/grsim/grsim_backend.cpp

```diff
@@ -1,11 +1,54 @@
-#include "grsim_backend.h"
+#include "backend_output/grsim/grsim_backend.h"
+#include <iostream>
+#include <optional>
 #include "ai/primitive/move_prim.h"
+#include "proto/grSim_Commands.pb.h"
+#include "proto/grSim_Packet.pb.h"
 
-GrSimBackend::GrSimBackend()
+using namespace boost::asio;
+
+GrSimBackend::GrSimBackend(std::string network_address, unsigned short port)
+    : network_address(network_address), port(port), socket(io_service)
+{
+    socket.open(ip::udp::v4());
+    remote_endpoint = ip::udp::endpoint(ip::address::from_string(network_address), port);
+}
+
+GrSimBackend::~GrSimBackend()
+{
+    socket.close();
+}
+
+void GrSimBackend::setRobotVelocities(
+    unsigned int robot_id, bool team_colour_yellow, Point velocity,
```

Comment:
We should make an enum for team color:
```
enum TeamColor { YELLOW, BLUE };
```
and then use that here (ie. `bool team_colour_yellow` -> `TeamColor team_color`)

Reply:
Will discuss at meeting today and fix in next PR that deals with interfaces